oxidation/crates/parsec_serialization_format/src/parser/field.rs

```diff
@@ -0,0 +1,76 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use miniserde::Deserialize;
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{Ident, Type};
+
+use super::utils::{inspect_type, quote_serde_as};
+use super::SerdeAttr;
+
+#[derive(Clone, Copy)]
+pub(crate) enum Vis {
+    Public,
+    Private,
+}
+
+#[derive(Deserialize)]
+pub(crate) struct Field {
+    name: String,
+    #[serde(rename = "type")]
+    ty: String,
+    introduced_in_revision: Option<u32>,
+}
+
+impl Field {
+    pub(crate) fn quote(&self, vis: Vis) -> TokenStream {
+        let rename = match self.name.as_str() {
+            "type" => Some("ty"),
+            _ => None,
+        };
+        let name: Ident =
+            syn::parse_str(rename.unwrap_or(&self.name)).unwrap_or_else(|_| unreachable!());
+        let raw_ty: Type = syn::parse_str(&self.ty).unwrap_or_else(|e| panic!("{e}"));
```

Comment:
I think it would be better to move the `syn::parse_str` into `inspect_type` function, and improve the error message to include the content of the field:
```rust
panic!("Invalid type value `{self.ty}`: {e}")
```

Reply:
Done :+1: